src/main/java/io/cdap/plugin/gcp/dataplex/source/config/DataplexBatchSourceConfig.java

```diff
@@ -0,0 +1,301 @@
+  }
+
+  @Nullable
+  public String getPartitionTo() {
+    return Strings.isNullOrEmpty(partitionTo) ? null : partitionTo;
+  }
+
+  @Nullable
+  public String getFilter() {
+    if (filter != null) {
+      filter = filter.trim();
+      if (filter.isEmpty()) {
+        return null;
+      }
+    }
+    return filter;
+  }
+
+  @Nullable
+  public Schema getSchema(FailureCollector collector) {
+    if (Strings.isNullOrEmpty(schema)) {
+      return null;
+    }
+    try {
+      return Schema.parseJson(schema);
+    } catch (IOException e) {
+      collector.addFailure("Invalid schema: " + e.getMessage(), null).
+        withConfigProperty(NAME_SCHEMA);
+    }
+    // if there was an error that was added, it will throw an exception, otherwise,
+    // this statement will not be executed
+    throw collector.getOrThrowException();
+  }
+
+  public FileFormat getFormat(String format) {
+    return FileFormat.from(format, FileFormat::canRead);
+  }
+
+  public Entity getAndValidateEntityConfiguration(FailureCollector collector, DataplexInterface dataplexInterface) {
+    IdUtils.validateReferenceName(referenceName, collector);
+    if (!Strings.isNullOrEmpty(location) && !containsMacro(NAME_LOCATION)) {
+      try {
+        dataplexInterface.getLocation(getCredentials(), tryGetProject(), location);
+      } catch (DataplexException e) {
+        configureDataplexException(location, NAME_LOCATION, e, collector);
+        return null;
+      }
+      if (!Strings.isNullOrEmpty(lake) && !containsMacro(NAME_LAKE)) {
+        try {
+          dataplexInterface.getLake(getCredentials(), tryGetProject(), location, lake);
+        } catch (DataplexException e) {
+          configureDataplexException(lake, NAME_LAKE, e, collector);
+          return null;
+        }
+
+        if (!Strings.isNullOrEmpty(zone) && !containsMacro(NAME_ZONE)) {
+          try {
+            dataplexInterface.getZone(getCredentials(), tryGetProject(), location, lake, zone);
+          } catch (DataplexException e) {
+            configureDataplexException(zone, NAME_ZONE, e, collector);
+            return null;
+          }
+          if (!Strings.isNullOrEmpty(entity) && !containsMacro(NAME_ENTITY)) {
+            try {
+              Entity entityBean = dataplexInterface.getEntity(getCredentials(), tryGetProject(), location,
+                lake, zone, entity);
+              return entityBean;
+            } catch (DataplexException e) {
+              configureDataplexException(entity, NAME_ENTITY, e, collector);
+              return null;
+            }
+          }
+        }
+      }
+    }
+    return null;
+  }
+
+  public void validateBigQueryDataset(FailureCollector collector, String project, String dataset, String tableName) {
+    Table sourceTable = BigQueryUtil.getBigQueryTable(project, dataset, tableName, this.getServiceAccount(),
+      this.isServiceAccountFilePath(), collector);
+    if (sourceTable == null) {
+      return;
```

Comment:
When can sourceTable be null? If it's null only in error scenarios, should we add a failure to the failure collector if we hit this case?

Reply:
Changes done.